src/lib/ModelConfig.cc

```diff
@@ -45,26 +42,250 @@ void ModelConfig::validate() {
   // Loop through expected fields and ensure a valid value exists
   std::vector<std::string> subFields;
   std::string root = "";
+  int validISA;
+
   // Core
   root = "Core";
-  subFields = {"Simulation-Mode", "Clock-Frequency", "Timer-Frequency",
-               "Micro-Operations", "Vector-Length"};
-  nodeChecker<std::string>(configFile_[root][subFields[0]], subFields[0],
+  subFields = {"ISA",
+               "Simulation-Mode",
+               "Clock-Frequency",
+               "Timer-Frequency",
+               "Micro-Operations",
+               "Vector-Length"};
+  validISA = nodeChecker<std::string>(
+      configFile_[root][subFields[0]], subFields[0],
+      std::vector<std::string>({"AArch64", "rv64"}), ExpectedValue::String);
+  nodeChecker<std::string>(configFile_[root][subFields[1]], subFields[1],
                            {"emulation", "inorderpipelined", "outoforder"},
                            ExpectedValue::String);
-  nodeChecker<float>(configFile_[root][subFields[1]], subFields[1],
+  nodeChecker<float>(configFile_[root][subFields[2]], subFields[2],
                      std::make_pair(0.f, 10.f), ExpectedValue::Float);
-  nodeChecker<uint32_t>(configFile_[root][subFields[2]], subFields[2],
+  nodeChecker<uint32_t>(configFile_[root][subFields[3]], subFields[3],
                         std::make_pair(1, UINT32_MAX), ExpectedValue::UInteger,
                         100);
-  nodeChecker<bool>(configFile_[root][subFields[3]], subFields[3],
+  nodeChecker<bool>(configFile_[root][subFields[4]], subFields[4],
                     std::make_pair(false, true), ExpectedValue::Bool, false);
-  nodeChecker<uint16_t>(configFile_[root][subFields[4]], subFields[4],
+  nodeChecker<uint16_t>(configFile_[root][subFields[5]], subFields[5],
                         {128, 256, 384, 512, 640, 768, 896, 1024, 1152, 1280,
                          1408, 1536, 1664, 1792, 1920, 2048},
                         ExpectedValue::UInteger, 512);
   subFields.clear();
 
+  // Check options that differ when different ISA's are used. First check ISA
+  // config option is valid before switching on it
+  if (validISA == 1) {
```

Comment:
1. Yes the node checker would flag this, however we need to read the value of the ISA option to be able to correctly check other options. Before reading the value, we check that the option contains valid values to prevent errors when reading
2. No else is provided as to allow other config options not reliant on the ISA to be checked. The node checker will still fail in this case as the ISA isn't correct
3. Other checks depend on the group mappings and other checks depend on those checks e.g. the port check depends on groupOptions_ and reservation stations depends on the ports being correct. Therefore, the RS check is dependent on the ports which is dependant on the ISA and so needs to be within the if